views/forgot.go

```diff
@@ -29,67 +29,65 @@ var notification = Notification{
+		userFromDB, err := v.user.GetUser(c.Request().Context(), u)
 		if err != nil {
 			// User doesn't exist, we'll pretend they've got an email
-			log.Printf("request for reset on unknown email \"%s\"", user1.Email)
+			log.Printf("request for reset on unknown email \"%s\"", userFromDB.Email)
 			return v.template.RenderTemplate(c.Response(), notification, templates.NotificationTemplate, templates.NoNavType)
 		}
 		url := uuid.NewString()
-		v.cache.Set(url, user1.UserID, cache.DefaultExpiration)
+		v.cache.Set(url, userFromDB.UserID, cache.DefaultExpiration)
+
+		mailer := mail.NewMailer(mail.Config{
+			Host:       v.conf.Mail.Host,
+			Port:       v.conf.Mail.Port,
+			Username:   v.conf.Mail.Username,
+			Password:   v.conf.Mail.Password,
+			DomainName: v.conf.Mail.DomainName,
+		})
 
 		// Valid request, send email with reset code
-		if v.mailer != nil {
-			v.mailer = mail.NewMailer(mail.Config{
-				Host:       v.conf.Mail.Host,
-				Port:       v.conf.Mail.Port,
-				Username:   v.conf.Mail.Username,
-				Password:   v.conf.Mail.Password,
-				DomainName: v.conf.DomainName,
-			})
-			if v.mailer == nil {
-				log.Printf("no Mailer present")
-				log.Printf("reset email: %s, code: %s, reset link: https://%s/reset?code=%s", user1.Email, url, v.conf.DomainName, url)
-				return v.template.RenderTemplate(c.Response().Writer, notification, templates.NotificationTemplate, templates.NoNavType)
+		if mailer != nil {
+			emailTemplate, err := v.template.RenderEmail(templates.ForgotEmailTemplate)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("failed to render email for forgot: %w", err))
 			}
 
 			file := mail.Mail{
 				Subject: "YSTV Security - Reset Password",
-				Tpl:     v.template.RenderEmail(templates.ForgotEmailTemplate),
-				To:      user1.Email,
+				Tpl:     emailTemplate,
+				To:      userFromDB.Email,
 				From:    "YSTV Security <no-reply@ystv.co.uk>",
 				TplData: struct {
 					Email string
 					URL   string
 				}{
-					Email: user1.Email,
-					URL:   "https://" + v.conf.DomainName + "/forgot/" + url,
+					Email: userFromDB.Email,
+					URL:   "https://" + v.conf.DomainName + "/reset/" + url,
 				},
 			}
 
-			err = v.mailer.SendMail(file)
+			err = mailer.SendMail(file)
 			if err != nil {
+				log.Printf("failed to send mail for forgot: %+v", err)
 				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("failed to send email for forgot: %w", err))
 			}
-			log.Printf("request for password reset email: \"%s\"", user1.Email)
+			_ = mailer.Close()
+
+			log.Printf("request for password reset email: \"%s\"", userFromDB.Email)
 		} else {
 			log.Printf("no Mailer present")
-			log.Printf("reset email: %s, code: %s, reset link: https://%s/reset?code=%s", user1.Email, url, v.conf.DomainName, url)
+			log.Printf("reset email: %s, code: %s, reset link: https://%s/reset/%s", userFromDB.Email, url, v.conf.DomainName, url)
```

Comment:
Food for thought: this could be considered sensitive info (as it'd allow anyone with the logs to reset the user's password), are we sure we want it to be stored and archived in the logs? (FWIW, I don't know what the right answer is, I can certainly see why it's useful.)

Reply:
It is mainly for the admin who is resetting the password, they can send them the link, I can probably keep it out of the logs though?

Reply:
That might be better - unlike the logs which are archived (in Loki) the page isn't.

Reply:
Though I may well be overthinking this 😅 your call

Reply:
I'll remove it from logs and just have requested for and by user